ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1800,17 +1800,11 @@ public Table getTable(final String dbName, final String tableName, String tableM
    * @throws LockException
    */
   private ValidWriteIdList getValidWriteIdList(String dbName, String tableName) throws LockException {
-    ValidWriteIdList validWriteIdList = null;
-    SessionState sessionState = SessionState.get();
-    HiveTxnManager txnMgr = sessionState != null? sessionState.getTxnMgr() : null;
-    long txnId = txnMgr != null ? txnMgr.getCurrentTxnId() : 0;
-    if (txnId > 0) {
-      validWriteIdList = AcidUtils.getTableValidWriteIdListWithTxnList(conf, dbName, tableName);
-    } else {
-      String fullTableName = getFullTableName(dbName, tableName);
-      validWriteIdList = new ValidReaderWriteIdList(fullTableName, new long[0], new BitSet(), Long.MAX_VALUE);
-    }
-    return validWriteIdList;
+    long txnId = Optional.ofNullable(SessionState.get())
+      .map(ss -> ss.getTxnMgr().getCurrentTxnId()).orElse(0L);
+    
+    return (txnId > 0) ? AcidUtils.getTableValidWriteIdListWithTxnList(conf, dbName, tableName) : 
+        new ValidReaderWriteIdList();
```

Comment:
new ValidReaderWriteIdList() ends up with a tableName=null instead of fullTableName, isn't this changing some behavior?

Reply:
if there is no txnId for an ACID table - that's wrong and we can return just a dummy